riscv/processor.h

```diff
@@ -234,8 +234,12 @@ class processor_t : public abstract_device_t
   bool extension_enabled_const(isa_extension_t ext) const {
     if (ext >= 'A' && ext <= 'Z')
       return state.misa->extension_enabled_const(ext);
-    else
-      return isa->extension_enabled(ext);  // assume this can't change
+    else {
+      return isa->extension_enabled(ext);
+    }
```

Comment:
The goal of `extension_enabled_const()` (versus plain `extension_enabled()`) is to document and enforce an assumption that an extension will not be dynamically changing during the life of the target program. A future change (such as this one, which allows some extensions to be newly disabled) could invalidate those assumptions, and without `extension_enabled_const()`, could be subtly broken.

When `--isa=rv64imafdc`, following a `csrw misa.C=0` it looks like `extension_enabled_const(EXT_ZCA)` will return `true`, even though this extension is not actually enabled anymore. It should always either fire an assertion or return the same value as `extension_enabled()`.

After this change, since `extension_enabled(EXT_ZCA)` can change during the program, we want `extension_enabled_const(EXT_ZCA)` to fire an assertion -- at least when C is enabled in the ISA (and I think it's okay to even when it's not).

Unfortunately I don't see a robust and easy way to do this without duplicating some of the logic in `misa_csr_t::unlogged_write()` about which extensions can be dynamically enabled.

We could keep track of which extensions have been assumed const by recording that in a new `bitset`, and keep track of which extensions have been dynamically disabled by recording that in another new `bitset`, and then assert in both `extension_enabled_const()` and `set_extension_enable()` by checking those new variables. That would be robust but it seems overly complicated.

Reply:
I've implemented this in [c9448044](https://github.com/scottj97/riscv-isa-sim/commit/c9448044bd96fe6a5666d68f90137fcbceb90a57) and it's not as bad as I thought.

The new assertions fire because in processor.cc we have this line:
```
  if (proc->extension_enabled_const(EXT_ZCMT))
    csrmap[CSR_JVT] = jvt = std::make_shared<jvt_csr_t>(proc, CSR_JVT, 0);
```

Which needs to be addressed first. If Zcmt is disabled (because `misa.C==0`) should the JVT CSR still be accessible? If yes, this should be changed to `extension_enabled()`. If not, then in addition to that change, `jvt_csr_t::verify_permissions()` needs to consider `extension_enabled(EXT_ZCMT)`.

Reply:
Nice job with c9448044bd96fe6a5666d68f90137fcbceb90a57: the dynamic approach is actually quite clean.

Re: `Zcmt`, our only choices are to make `misa.C` read-only when Zcmt is present, or make the change you suggested.  It's more in keeping with this PR's approach to do the latter.  I'll open a new PR since I don't have write permissions on this one.

Reply:
OK. I'll close this PR.